Add "Export..." entry to the menu of UMLet standalone (Swing edition)…

frademacher opened the pull request:
… for a quicker export of the current diagram

The changes in this commit extend the menu with an "Export..." entry. It has its mnemonic and shortcut set to "E" and Ctrl+E, respectively. Activating the menu entry will result in showing the file chooser to the user. Depending on the current "export status" of the current diagram, the file chooser is initialized differently:

1. If the diagram has not been exported before: Path and filename in the chooser are the same as if the diagram would have been exported with the already existing "Export as..." menu entry. The extension in the chooser will be set to the extension specified in the "last_export_format" config entry. In case such an entry does not exist or contains a value that is not an export format supported by UMLet, the extension will default to "PDF".
    
2. If the diagram was exported before, the file chooser will be initialized with the directory, filename and extension of the previously exported file. Note, that a previous export may be also the result of an execution of the already existing "Export as..." menu entry.

afdia commented on umlet-swing/src/main/java/com/baselet/gui/ExportHandler.java:
```diff
@@ -0,0 +1,113 @@
+	private String filename(String filePath) {
+		File fileObject = new File(filePath);
+		String filename = fileObject.getName();
+		int extensionPos = filename.indexOf(".");
```
should probably use lastIndexOf

afdia commented on umlet-swing/src/main/java/com/baselet/gui/ExportHandler.java:
```diff
@@ -0,0 +1,113 @@
+	}
+
+	private String extension(String filePath) {
+		int extensionPos = filePath.indexOf(".");
```
should probably use lastIndexOf

afdia commented:
Thanks for the contribution, I have commented 2 parts where I would use lastIndexOf instead of indexOf to match the last . in the filename instead of the first one (which could be problematic with names such as my.file.pdf)

And is there a reason why you chose to treat "String directory, String filename" as 2 separate parameters which are passed through instead of keeping it as a single "String filepath" parameter which is split within the DiagramFileHandler.createSaveFileChooser method?
I'm just asking because it would probably a bit easier to understand and use if filepath is only split when necessary and not sooner, but I guess your approach is also ok.

Other than that it looks fine and it works as expected, so please implement the minor change of lastIndexOf and let me know what you think about the filepath approach and I can merge the pull request

frademacher commented:
No problem, glad that you consider its inclusion in upstream UMLet. I changed both `filename()` and `directory()` to use `lastIndexOf()` rather than `indexOf()`, which totally makes sense.

I also changed the implementation of `ExportHandler` to pass the filepath, i.e., a String constisting of directory and filename, to the `DiagramHandler` instead of splitting it first, which was unnecessary in `ExportHandler`. If I recall correctly, this behavior originated from my first implementation proposal that contained the button in the menu panel.

However, I came to recognize that the first commit [2b6c4aa](https://github.com/frademacher/umlet/commit/2b6c4aa6a5fd57f7e016a4669b7b3c6b4fe59df8), which introduces the current implementation, lacks the "#501" issue pointer at its beginning. If you would like to receive an updated PR with a cleaner commit history, just let me know. I would then also squash both last commits [8af3510](https://github.com/frademacher/umlet/commit/8af3510fc64c22d67265faece7109c069ca58695) and [44f84af](https://github.com/frademacher/umlet/commit/44f84afd5df45b803504d8d67f49ee35aa562c56) into the introductory one?

afdia commented:
Good idea, please add the issue prefix to the commit message and squash the commits to a single one, it will help other developers in the long run

Thanks!

frademacher commented:
Okay, I polished the commit history as mentioned and did a force push. It seems to me that GitHub recognized everything correctly. In case you have no additional remarks, I would consider the implementation of the quick export feature being finished.
